src/Test/Utilities/CLRHelpers.cs

```diff
@@ -64,12 +65,14 @@ public static string[] PeVerify(string filePath)
             return PeVerify(File.ReadAllBytes(filePath), AppDomain.CurrentDomain.Id, filePath);
         }
 
-        private static readonly object Guard = new object();
+        private static readonly Mutex s_guard = new Mutex(false, new Guid().ToString());
```

Comment:
I couldn't find a synchronization routine more course grained than named mutex that locks across app-domain. Suggestions?